pkg/cmd/time-entry/util/report.go

```diff
@@ -91,18 +103,44 @@ func PrintTimeEntry(
 	b := config.GetBool(cmdutil.CONF_SHOW_TOTAL_DURATION)
 	config.SetBool(cmdutil.CONF_SHOW_TOTAL_DURATION, false)
 
-	err := PrintTimeEntries(ts, out, config, of)
+	err := PrintTimeEntries(updateTimeZone(ts, of), out, config, of)
 
 	config.SetBool(cmdutil.CONF_SHOW_TOTAL_DURATION, b)
 
 	return err
 }
 
+func updateTimeZone(tes []dto.TimeEntry, of OutputFlags) []dto.TimeEntry {
+
+	if of.TimeZone == "" {
+		return tes
+	}
+
+	var loc *time.Location
+
+	if of.TimeZone == "local" || of.TimeFormat == "" {
+		loc = time.Local
+	} else {
+		// parses of.TimeZone as a time.Location
+		loc, _ = time.LoadLocation(of.TimeZone)
+	}
```

Comment:
since you checked before you don't need to check again
```suggestion
func updateTimeZone(tes []dto.TimeEntry, of OutputFlags) []dto.TimeEntry {
	if of.TimeZone == "" {
		return tes
	}

	var loc *time.Location
	if of.TimeZone == "local" {
		loc = time.Local
	} else {
		// parses of.TimeZone as a time.Location
		loc, _ = time.LoadLocation(of.TimeZone)
	}
```